app-frontend/src/app/services/projects/project.service.js

```diff
@@ -148,6 +148,55 @@ export default (app) => {
             return this.Project.query(params).$promise;
         }
 
+        searchQuery() {
+            let deferred = this.$q.defer();
```

Comment:
Use `this.$q((resolve, reject) => {...})`
I detailed reasons why in my PR

Reply:
:+1: